src/t8_forest/t8_forest_balance.cxx

```diff
@@ -81,34 +81,39 @@ t8_forest_balance_adapt (t8_forest_t forest, t8_forest_t forest_from, const t8_l
     for (iface = 0; iface < num_faces; iface++) {
       /* Get the element class and scheme of the face neighbor */
       neigh_class = t8_forest_element_neighbor_eclass (forest_from, ltree_id, element, iface);
-      /* Allocate memory for the number of half face neighbors */
-      num_half_neighbors = scheme->element_get_num_face_children (tree_class, element, iface);
-      half_neighbors = T8_ALLOC (t8_element_t *, num_half_neighbors);
-      scheme->element_new (neigh_class, num_half_neighbors, half_neighbors);
-      /* Compute the half face neighbors of element at this face */
-      neighbor_tree = t8_forest_element_half_face_neighbors (forest_from, ltree_id, element, half_neighbors,
-                                                             neigh_class, iface, num_half_neighbors, NULL);
-      if (neighbor_tree >= 0) {
-        /* The face neighbors do exist, check for each one, whether it has
+      if (neigh_class != T8_ECLASS_INVALID) {
```

Comment:
I only added this line and the "if (neighbor_tree >= 0) {" below.
The rest is just indentation.